src/backend/marsha/core/tests/test_models_video.py

```diff
@@ -93,3 +96,53 @@ def test_models_video_is_ready_to_show(self):
             video = VideoFactory(live_state=live_choice[0], live_type=RAW)
 
             self.assertEqual(video.is_ready_to_show, True)
+
+    def test_models_video_is_scheduled_none(self):
+        """Checks that with a starting_at at None video it's not in the scheduled mode."""
+        # With no starting_date, video is not scheduled
+        video = VideoFactory(starting_at=None)
+        self.assertFalse(video.is_scheduled)
+        self.assertEqual(video.live_state, None)
+
+    def test_models_video_is_scheduled_with_state(self):
+        """The property is_scheduled is set according to starting_at and live_state."""
+        video = VideoFactory(starting_at="2100-01-01 10:00:00")
+        # Video is in the scheduled mode
+        self.assertTrue(video.is_scheduled)
+        self.assertEqual(video.live_state, None)
+
+        # With any live_state other than None, is_scheduled is False
+        for live_choice in LIVE_CHOICES:
+            video = VideoFactory(
+                live_state=live_choice[0],
+                live_type=RAW,
+                starting_at="2100-01-01 10:00:00",
+            )
+
+            self.assertFalse(video.is_scheduled)
+
+    def test_models_video_is_scheduled_with_date_constraint(self):
+        """
+        Testing the switch of is_scheduled with time over now.
+
+        The property is_scheduled is set according to starting_at. It gets automaticaly
+        updated to not scheduled if date is over.
+        """
+        # Can't set with a date in the past
+        with self.assertRaises(ValidationError) as error:
+            VideoFactory(starting_at="2000-01-01 10:00:00")
+
+        self.assertEqual(
+            error.exception.messages,
+            ["2000-01-01 10:00:00 is not a valid date, date should be planned after!"],
+        )
+        # Set to now plus 1 second and wait
+        video = VideoFactory(starting_at=timezone.now() + timedelta(seconds=1))
+        # Video is scheduled
+        self.assertTrue(video.is_scheduled)
+        self.assertTrue(video.get_ready_clause)
```

Comment:
`get_ready_clause` is a static method, not a property. Moreover the method return a `Q` object, not a boolean. I don't know what you are testing here but it's wrong.
The `get_ready_clause` behaviour is tested through `lti.get_or_create_resource` and all the tests are in the `test_lti.py` module. And you must add test to this module to test your modification.